gc3apps/bf.uzh.ch/gIdRiskUML.py

```diff
@@ -169,7 +169,7 @@ def new_tasks(self, extra):
         # Copy base dir
         localBaseDir = os.path.join(os.getcwd(), 'localBaseDir')
 #        gc3libs.utils.copytree(self.params.initial, '/mnt/shareOffice/ForwardPremium/Results/sensitivity/wGridSize/dfs')
-        gc3libs.utils.copytree(self.params.initial, localBaseDir)
+        shutil.copytree(self.params.initial, localBaseDir)
```

Comment:
no, i'm not 100% sure yet.

whenever utils.copytree was used, it only ever had src and dest arguments, so at very least the unused kwargs can go.

`shutils.copytree` also creates the dest dir. Docs say: `"it will be created as well as missing parent directories"`. as far as i can tell, they are the same in this respect?

the only difference is the 'only changed' functionality. if this functionality is required, i would prefer replace utils.copytree with something simpler and without unused arguments like `copy_changed_files(src, dest)`?

IMO using the same name as `shutils`, and then using both functions throughout the module, is difficult to understand and could cause namespace issues if imports are changed a bit (i.e. `from shutil import copytree`)

Reply:
Looking more at shutil docs, we could accomplish `only_changed` behaviour by passing in something smart as the `ignore` param of `shutil.copytree`. That something smart may actually need to be a function of a few lines. So, in utils, we would just need:

```python
def only_changed(*args):
    return some_logic
````

And where utils.copytree was being used:

```python
from utils import only_changed
shutil.copytree(src, dest, ignore=only_changed)
```

Basically, the standard library can handle everything we need, so we should use it instead of rolling our own :)

Reply:
> using the same name as `shutils`, and then using both functions throughout the module, is difficult to understand and could cause namespace issues if imports are changed a bit (i.e. `from shutil import copytree`)

I've always added variants of stdlib functions to `utils.py` under the same name to signal that they were indeed variants, but this is a good argument against.

Reply:
> Basically, the standard library can handle everything we need, so we should use it instead of rolling our own :)

Totally agree; please go ahead with the changes if they mean we can rely more on stdlib than on our code.

Reply:
Great. I'm always pleased to reduce size of codebase :)

Reply:
Update on this, it is not easy to use the `ignore` kwarg as I'd hoped. But it seems that keyword arguments in `utils.copytree` as currently implemented do nothing anyway. They are recursively passed for subdirectories, but are never passed to `copyfile`, so the defaults of copyfile are always used.

So the decision for me is: replace `utils.copytree` with something like `copy_changed_files`, and have both this and `copyfile` take just `src` and `dest` arguments, so that we can remove most of their unused branches.

Reply:
OK with that; actually if the "changed files" feature is never used, we can just delete `gc3libs.utils.copytree()` and replace it with `shutils.copytree()`

Reply:
Removed them. Could not find any instance of their special kwargs being used

Reply:
Good, thanks!